prow/github/client.go

```diff
@@ -687,8 +688,15 @@ func NewClientFromOptions(fields logrus.Fields, options ClientOptions) (TokenGen
 	options = options.Default()
 
 	// Will be nil if github app authentication is used
+	// Hack: We manually retrieve personal access token to support some operation missing GitHubApp jwtToken.
+	// This helps avoid situation that tide calls github api anonymously and leads to fields missing
 	if options.GetToken == nil {
-		options.GetToken = func() []byte { return nil }
+		personalAccessToken := os.Getenv("GITHUB_TOKEN")
```

Comment:
/hold

We should not mix apps aush and PAT auth. There might be a bug in the area, but this is not a reasonable solution.

Reply:
I agree. Is there a better way that I can help with from your perspective?

Reply:
Technically, I feel app temp token should be generated and renewed, so we can return that credential in `GetToken`. I probably need more time to dig into the code and get familiar with app authentication

Reply:
As mentioned in https://github.com/kubernetes/test-infra/issues/24141#issuecomment-1016624541, the issue is likely that your app has insufficient permissions, it needs `Contents: Read & Write`. Apps auth definitely does get an Authorization Header set, it just happens on a different layer in a http transport, as it needs to inspect the request in order to decide what token to set.